src/main/java/org/jenkins_ci/plugins/flexible_publish/ConditionalPublisher.java

```diff
@@ -169,20 +170,41 @@ public ConditionalPublisherDescriptor getDescriptor() {
     }
 
     public boolean prebuild(final AbstractBuild<?, ?> build, final BuildListener listener) {
-        for (BuildStep publisher: getPublisherList()) {
-            if (!runner.prebuild(condition, publisher, build, listener)) {
-                return false;
-            }
+        if (getPublisherList().size() <= 0) {
+            return true;
+        } else if (getPublisherList().size() > 1) {
+            return runner.prebuild(condition, new RunAllBuilder(getPublisherList()), build, listener);
         }
-        return true;
+        return runner.prebuild(condition, getPublisherList().get(0), build, listener);
     }
 
     public boolean perform(final AbstractBuild<?, ?> build, final Launcher launcher, final BuildListener listener)
                                                                                                 throws InterruptedException, IOException {
-        for (BuildStep publisher: getPublisherList()) {
-            if (!runner.perform(condition, publisher, build, launcher, listener)) {
+        if (getPublisherList().size() <= 0) {
+            return true;
+        } else if (getPublisherList().size() > 1) {
+            return runner.perform(condition, new RunAllBuilder(getPublisherList()), build, launcher, listener);
```

Comment:
This is a bug in your head, you can't just break all installations because "fail first" is a bug for you. FIY there are many different strategies of execution, for example read `mvn -help`
```
 -fae,--fail-at-end                     Only fail the build afterwards;
                                        allow all non-impacted builds to
                                        continue
 -ff,--fail-fast                        Stop at first failure in
                                        reactorized builds
 -fn,--fail-never                       NEVER fail the build, regardless
                                        of project result
```